### PR TITLE
added logic.js file and random table filling

### PR DESCRIPTION
**Pull request by Ivan-Andr**
I added some chenges to random table filling, it works but positions of red and black numbers are same as in our first table.

### CODE REVIEW

**Comment by levnikan** [comment]
I think the algorithm is correct, but initial way we set colors is wrong. The destribution of colors in not equal. Alternative solution could be:
1) create an array of black objects [{index: 1, color:"black"},{index: 2, color:"black"}, ... {index: 25, color:"black"}], 
2) create an array of red objects [{index: 1, color:"red"}, {index: 2, color:"red"}, ... {index: 24, color:"red"}] 
3) concatenate those two arrays into one 
4) shuffle the final array
And then we will simply check the color value of each object upon iteration: tableArr[k].color === "red" ? cell.className = "table_cell_red" : cell.className = "table_cell_black"

File: script.js [hunk 1]
```diff
@@ -6,3 +6,27 @@ table();
 buttons();
 startApp();
 changeLanguage();
+
+const tableArr2 = new Array(49);
```
Comment (levnikan) [inline script.js]: why do you have same code here and below?

File: src/ui/table.js [hunk 2]
```diff
@@ -1,23 +1,47 @@
...
+// ];
+
+// const redArray = [
+//   2, 5, 6, 7, 10, 11, 13, 15, 16, 19, 21, 25, 27, 28, 29, 31, 32, 37, 40, 42,
+//   43, 45, 46, 48,
+// ];
+const tableArr2 = new Array(49);
+const redArr2 = [];
+tableArr2.fill(-1);
+function arrayFill(startElem) {
```
Comment (levnikan) [inline src/ui/table.js]: it's better to create 2 pure functions (one for table and one for red) and make them independent from the outer scope e.g. passing array as parameter and then return filled one.

**Comment by Ivan-Andr** [comment]
I Added your version of array filling. But I can't get updated testNumber after form filling. I want to use it like starting value of red and black number arrays. 
`// let startElem = userObject.tableNumber;
// console.log(startElem, typeof startElem);`

File: src/ui/table.js [hunk 3]
```diff
@@ -10,28 +10,48 @@
...
+//     redArr2.push(n);
+//   }
+// }
+// //to start test with 1 - arrayFill(1), to start test with 2 - arrayFill(2) etc.
+// arrayFill(2);
+///////var2////////////
+let blackRedArr = [];
+// let startElem = userObject.tableNumber;
+// console.log(startElem, typeof startElem);
+function arrayFill(startEl) {
```
Comment (levnikan) [inline src/ui/table.js]: I would call it `getBlackRedArray` and move into utils folder, same for shuffle, let's call it `getShuffledArray`. You can also add description to those functions if you wish https://stackoverflow.com/questions/73219300/how-can-i-add-a-description-to-a-function-in-javascript

File: src/ui/table.js [hunk 4]
```diff
@@ -10,28 +10,48 @@
...
-tableArr2.fill(-1);
-function arrayFill(startElem) {
-  for (let i = startElem; i <= 24 + startElem; i++) {
-    let n = Math.floor(Math.random() * 49);
-    while (tableArr2[n] >= 0) {
-      n = Math.floor(Math.random() * 49);
-    }
-    tableArr2[n] = i;
+
+//////////////////var1/////////////
```
Comment (levnikan) [inline src/ui/table.js]: let's clean all the comments

**Review by levnikan — CHANGES_REQUESTED** [review]

**Comment by levnikan** [comment]
> I Added your version of array filling. But I can't get updated testNumber after form filling. I want to use it like starting value of red and black number arrays. `// let startElem = userObject.tableNumber; // console.log(startElem, typeof startElem);`

you can try to add 2 more steps to the event listener https://github.com/Ivan-Andr/psychological-test/blob/issue%2311/src/ui/form.js#L51 1) delete the existing table and its children 2) generate it again and pass the number entered like table(numberOfTheTest)

File: src/ui/form.js [hunk 5]
```diff
@@ -43,13 +44,14 @@ export const form = function () {
...
     userObject.tableNumber = inputTestNumber.value;
   };
-  inputDataForm.addEventListener("submit", function (e) {
+  inputDataForm.addEventListener('submit', function (e) {
     e.preventDefault();
     submitModalForm();
     closeModalForm();
     updateUserObject();
-    console.log(userObject);
+    table(shuffledArray(getBlackRedArr(Number(userObject.tableNumber))));
```
Comment (levnikan) [inline src/ui/form.js:52]: haha, that's a bit too much, better to write a couple of lines declaring new variables, but let's leave it for now

**Review by levnikan — DISMISSED** [review]

File: src/ui/buttons.js [hunk 6]
```diff
@@ -39,13 +46,17 @@ export const buttons = () => {
...
-    ) {
+    let message = "";
+    if (langSelect.value === "ru") {
+      message = langArr["confirm-window-message"]["ru"];
+    } else if (langSelect.value === "uz") {
+      message = langArr["confirm-window-message"]["uz"];
+    } else {
+      message = langArr["confirm-window-message"]["en"];
+    }
+    if ((confirmWindow = window.confirm(`${message}`))) {
```
Comment (levnikan) [inline src/ui/buttons.js:57]: better would be to make it with the modal window, but let's proceed for now and create a tech debt ticket to improve it.

**Review by levnikan — APPROVED** [review]

**Pull request merged**